tremor-script/src/ast/query.rs

```diff
@@ -298,6 +298,21 @@ pub(crate) enum GroupByInt<'script> {
     },
 }
 
+pub(crate) trait GroupByVisitor<'script> {
```

Comment:
since the visitors are pretty large and quite distinct from the rest of the at code,  can we pull them out into `ast/visitors.rs`?